client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -144,13 +155,38 @@
         List<ByteBuffer> data = events.stream().map(serializer::serialize).collect(Collectors.toList());
         CompletableFuture<Void> ackFuture = new CompletableFuture<Void>();
         synchronized (writeFlushLock) {
-            synchronized (writeSealLock) {
-                SegmentOutputStream segmentWriter = getSegmentWriter(routingKey);
-                segmentWriter.write(PendingEvent.withHeader(routingKey, data, ackFuture));
+            if (data.stream().mapToInt(m -> m.remaining()).sum() > Serializer.MAX_EVENT_SIZE) {
+                writeLargeEvent(routingKey, data, ackFuture);
```

Comment:
There is nothing preventing the retransmit of inflight events from sealed segments when large events are written right?
Would'nt this change the order of events?

Reply:
The first like of `writeLargeEvent` calls `flush()` once that completes there should be no events to retransmit.